### PR TITLE
Added support for model autocomplete filter

### PR DESCRIPTION
**Pull request by pulzarraider**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes, but minor (missing and few new methods added to interfaces, changed CSS class in sonata_type_model_autocomplete)
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2360, #2311
| License       | MIT
| SonataDoctrineORMAdmin PR | sonata-project/SonataDoctrineORMAdminBundle#388

Example:
```php
protected function configureDatagridFilters(DatagridMapper $datagridMapper)
{
    $datagridMapper
        ->add('category', 'doctrine_orm_model_autocomplete', array(), null, array(
            'property'=>'title',
        ))
    ;
}
```

### CODE REVIEW

**Comment by rande** [comment]
@pulzarraider you need to rebase this PR

File: Controller/HelperController.php [hunk 1]
```diff
@@ -439,4 +454,35 @@ private function retrieveFieldDescription(AdminInterface $admin, $field)
...
+    {
+        $admin->getFilterFieldDescriptions();
+
+        $fieldDescription = $admin->getFilterFieldDescription($field);
+
+        if (!$fieldDescription) {
+            throw new \RuntimeException(sprintf('The field "%s" does not exist.', $field));
+        }
+
+        if ($fieldDescription->getType() !== 'doctrine_orm_model_autocomplete') {
```
Comment (rande) [inline Controller/HelperController.php]: This value must be configured somehow, maybe with parameters

**Comment by cassianotartari** [comment]
:+1: I would love to have this filter type, need it for a neighbor and city filters. Did an update now with ``"sonata-project/admin-bundle": "~2.3"`` and still not implemented. Many thanks!

**Comment by rande** [comment]
@cassianotartari this feature will included in a 2.4 release

**Comment by cassianotartari** [comment]
thanks @rande, so maybe I will need to wait more or implemented mine, huge project to update.

File: Controller/HelperController.php (hunk 1 above)
Comment (pulzarraider) [inline Controller/HelperController.php]: This is just a small check. I think that  it is not needed to add it to configuration. I have removed it to not include doctrine orm inside this universal admin.

**Comment by pulzarraider** [comment]
PR rebased

**Comment by lsmith77** [comment]
ready to go into 2.4?

**Comment by rande** [comment]
@pulzarraider is this PR ready ? if so just press the merge button

**Comment by pulzarraider** [comment]
I will test it once again to be sure in the weekend.

**Comment by lsmith77** [comment]
ping

**Comment by pulzarraider** [comment]
Thanks to everyone who helped me finish this PR, especially @webdevilopers and @krewetka for reporting bugs.

**Pull request merged**